File: week02/ee/더하기사이클.py
N = int(input())

cycle = 0

prev = N
pres = 0

while True :
    if prev < 10:
        pres = str(prev) + str(prev)
    else :
        pres = str(prev%10) + str(((prev % 10) +(prev//10)) % 10)
    prev = int(pres)
    
    cycle += 1
    if prev == N:
        break

print(cycle)

File: week02/ee/더하기사이클.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "dev/stdin" : "./input.txt";

const N = parseInt(fs.readFileSync(filePath).toString().trim());

let cycle = 0

let a = Math.floor(N / 10)
let b = N % 10

let prev = 0

while (true) {
  prev = b * 10 + (a + b) % 10

  a = Math.floor(prev / 10)
  b = prev % 10

  cycle ++;
  if (prev === N) break;
}

console.log(cycle)
